src/aiq/utils/request_utils.py

```diff
@@ -0,0 +1,246 @@
+import json
+import logging
+import re
+import urllib.parse
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+from aiq.authentication.exceptions.request_exceptions import BaseUrlValidationError
+from aiq.authentication.exceptions.request_exceptions import BodyValidationError
+from aiq.authentication.exceptions.request_exceptions import HTTPHeaderValidationError
+from aiq.authentication.exceptions.request_exceptions import HTTPMethodValidationError
+from aiq.authentication.exceptions.request_exceptions import QueryParameterValidationError
+from aiq.data_models.authentication import HTTPMethod
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPRequestValidator:
+    """
+    HTTP request validation utility class.
+
+    Provides comprehensive validation for HTTP request components including URLs, methods,
+    headers, query parameters, and body data. Supports flexible input parsing for both
+    JSON strings and dictionaries.
+    """
+
+    @staticmethod
+    def parse_flexible_input(input_data: str | dict | None, input_name: str) -> Optional[Dict[str, Any]]:
```

Comment:
This is a lot of validation were adding that would already belong in `httpx`, right?